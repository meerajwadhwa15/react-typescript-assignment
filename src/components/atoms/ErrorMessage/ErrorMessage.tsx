import React, { FunctionComponent, ReactNode } from 'react'

export type Props = {
  children: ReactNode
}

const ErrorMessage: FunctionComponent<Props> = ({ children }) => (
  <div>{children}</div>
)

export default ErrorMessage
