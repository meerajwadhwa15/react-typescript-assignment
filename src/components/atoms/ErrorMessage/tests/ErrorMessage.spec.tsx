import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import ErrorMessage, { Props } from '../ErrorMessage'

const props: Props = {
  children: 'test',
}

describe('ErrorMessage Enzyme testing', () => {
  test('renders without crashing', () => {
    shallow(<ErrorMessage {...props} />)
  })
})

describe('ErrorMessage Snapshot testing', () => {
  test('Matches the snapshot', () => {
    const component = create(<ErrorMessage {...props} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
