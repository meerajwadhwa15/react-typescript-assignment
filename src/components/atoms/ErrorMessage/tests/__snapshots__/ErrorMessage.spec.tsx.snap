// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage Snapshot testing Matches the snapshot 1`] = `
<div>
  test
</div>
`;
